tfsdk/attribute.go

```diff
@@ -369,3 +376,42 @@ func (a Attribute) validate(ctx context.Context, req ValidateAttributeRequest, r
 		}
 	}
 }
+
+// modifyPlan runs all AttributePlanModifiers
+func (a Attribute) modifyPlan(ctx context.Context, req ModifyAttributePlanRequest, resp *ModifyAttributePlanResponse) {
+	attrConfig, diags := req.Config.GetAttribute(ctx, req.AttributePath)
+	resp.Diagnostics = append(resp.Diagnostics, diags...)
+	if diagnostics.DiagsHasErrors(diags) {
+		return
+	}
+	req.AttributeConfig = attrConfig
+
+	attrState, diags := req.State.GetAttribute(ctx, req.AttributePath)
+	resp.Diagnostics = append(resp.Diagnostics, diags...)
+	if diagnostics.DiagsHasErrors(diags) {
+		return
+	}
+	req.AttributeState = attrState
+
+	attrPlan, diags := req.Plan.GetAttribute(ctx, req.AttributePath)
+	resp.Diagnostics = append(resp.Diagnostics, diags...)
+	if diagnostics.DiagsHasErrors(diags) {
+		return
+	}
+	req.AttributePlan = attrPlan
+
+	modifyReq := ModifyAttributePlanRequest{
+		AttributePath:   req.AttributePath,
+		Config:          req.Config,
+		State:           req.State,
+		Plan:            req.Plan,
+		AttributeConfig: req.AttributeConfig,
+		AttributeState:  req.AttributeState,
+		AttributePlan:   req.AttributePlan,
+		ProviderMeta:    req.ProviderMeta,
+	}
+	for _, planModifier := range a.PlanModifiers {
+		planModifier.Modify(ctx, modifyReq, resp)
+		modifyReq.AttributePlan = resp.AttributePlan
+	}
```

Comment:
Is it worth making this recursive, like `Attribute.validate` is, so it goes through all the nested attributes of this attribute and calls their `modifyPlan` method, too?